Documentation/storage.md

```diff
@@ -0,0 +1,161 @@
+# Storage options
+
+Dex requires persisting state to perform various tasks such as track refresh tokens, preventing replays, and rotating keys. This document is a summary of the storage configurations supported by dex.
+
+Regardless of the implementation, dex saves sensitive data in its backing storage: signing keys, bcrypt'd passwords, etc. Storage breaches are serious because attacks may propagate to any application which relies on dex. Whatever storage option used, transport security and database ACLs are both musts.
+
+## Caveat: Running replicated instances
+
+Validation is still being done to ensure correctness when multiple instances of dex are using the same backing storage. While everything "should just work", edge cases require time to investigate and write tests for. This work can be followed on the issue tracker [here](https://github.com/coreos/dex/issues/600).
+
+For now, instances should run a single instance of dex and occationally run two during upgrades.
```

Comment:
I would reword this to be (if it's accurate):

The Dex team suggests running **one dex instance most of the time** and **two instance during upgrades**.